contracts/RelayHub.sol

```diff
@@ -208,6 +215,7 @@ contract RelayHub is RelayHubApi {
         bool success_post;
         bytes memory ret;
         (success_post,ret) = address(this).call(abi.encodeWithSelector(this.recipient_calls.selector,from,to,msg.sender,encoded_function,transaction_fee,gas_limit,initial_gas));
+        nonces[from]++;
         // Relay transaction_fee is in %.  E.g. if transaction_fee=40, payment will be 1.4*used_gas.
         uint charge = (gas_overhead+initial_gas-gasleft())*gas_price*(100+transaction_fee)/100;
```

Comment:
charge is calculated based on `gas_overhead`, which we try to make as static as possible.
but the next code (starting with `if (!success_ports)` is conditional, and doesn't have a fixed gas.
Should move all param calculations before the `charge` calculation, (except the `require`... its conditional, but in normal case it always succeeds)

BTW: maybe this is the cause our estimation of gas_overhead where broken: the `keccak` has different gas usage depending on the function call length.

Reply:
done